docs/userguide/searchspace.md

```diff
@@ -1,46 +1,271 @@
+Note that a search space is not necessarily equal to the space of allowed measurements. That is, if configured properly, it is possible to add measurements to a campaign that are not part of the search space.
+
+In BayBE, a search space is a union of two (potentially empty) subspaces. The [`SubspaceDiscrete`](baybe.searchspace.discrete.SubspaceDiscrete) contains all discrete parameters, i.e., all parameters which inherit from [`DiscreteParameter`](baybe.parameters.base.DiscreteParameter). The [`SubspaceContinuous`](baybe.searchspace.continuous.SubspaceContinuous) contains all continuous parameters, i.e., all parameters which inherit from [´ContinuousParameter´](baybe.parameters.base.ContinuousParameter).
+
+Depending on which of the subspaces are non-empty, a `SearchSpace` has exactly one of the three following [`SearchSpaceType`](baybe.searchspace.core.SearchSpaceType):
+1. If `SubspaceDiscrete` is non-empty but `SubspaceContinuous` is empty, the `SearchSpace` is *discrete* and has the type [`SearchSpaceType.DISCRETE`](baybe.searchspace.core.SearchSpaceType.DISCRETE).
+2. If `SubspaceDiscrete` is empty but `SubspaceContinuous` is non-empty, the `SearchSpace` is *continuous* and has the type [`SearchSpaceType.CONTINUOUS`](baybe.searchspace.core.SearchSpaceType.CONTINUOUS).
+3. If both subspaces are non-empty, the `SearchSpace` is *hybrid* and has the type [`SearchSpaceType.HYBRID`](baybe.searchspace.core.SearchSpaceType.HYBRID).
+
+## `SubspaceDiscrete`
+
+The `SubspaceDiscrete` contains all the discrete parameters of a `SearchSpace`. There are different ways of constructing this subspace.
+
+### Building from the Product of Parameter Values
+
+The method [`SearchSpace.from_product`](baybe.searchspace.discrete.SubspaceDiscrete.from_product) constructs the full cartesian product of the provided parameters:
+
+```python
+from baybe.parameters import NumericalDiscreteParameter
+from baybe.searchspace import SubspaceDiscrete
+
+parameters = [
+    NumericalDiscreteParameter(name="x0", values=[1, 2, 3]),
+    NumericalDiscreteParameter(name="x1", values=[4, 5, 6]),
+    NumericalDiscreteParameter(name="x2", values=[7, 8, 9]),
+]
+subspace = SubspaceDiscrete.from_product(parameters=parameters)
+```
+
+In this example, `subspace` has a total of 27 different parameter configuration.
+
+~~~
+       x0   x1   x2
+  0   1.0  4.0  7.0
+  1   1.0  4.0  8.0
+  2   1.0  4.0  9.0
+  ..  ...  ...  ...
+  24  3.0  6.0  7.0
+  25  3.0  6.0  8.0
+  26  3.0  6.0  9.0
+  
+  [27 rows x 3 columns]
+~~~
+
+### Constructing from a dataframe
+
+[`SubspaceDiscrete.from_dataframe`](baybe.searchspace.discrete.SubspaceDiscrete.from_dataframe) constructs a discrete subspace from a given dataframe.
+This method tries to infer the parameter types during the creation of the subspace.
+The parameter type is only necessary if special parameters should be used.
```

Comment:
your copying my colloquial chat wording a bit too exaclty here :D the proper explanation says somethign like `tries to indentify a data column as numerical discrete parameter (unless specified otherwise by the parameters list) and sues a cetegorical parameter as fallback.`

I see the sentence below, but it still odesnt mention the fallback

in other words: if your columns are numerical and you intend them to be discrete num params you dont have to do anything, if any param type except cat is intended it needs to be part of the parameter list, anything not covered in the previous words will be cat

Reply:
Changed the descriotion, please resolve if this is now better.

Reply:
gucci